lua/nvim-treesitter/query_predicates.lua

```diff
@@ -87,3 +87,10 @@ end)
 
 -- Just avoid some anoying warnings for this directive
 query.add_directive('make-range!', function() end)
+
+query.add_directive('downcase!', function(match, _, bufnr, pred, metadata)
+  local node = match[pred[2]]
+  local text = query.get_node_text(node, bufnr)
+  local language = string.lower(text)
+  metadata.language = language
+end)
```

Comment:
Although I agree with the implementation, it may be better to make it a bit more generic, maybe by providing the key to set instead of just using `language` this would make it look a bit like set, and may feel a little better.

Reply:
@DerekStride vigoux is right. Could you add an argument that determines which metadata to downcase?

Reply:
It could even become a modifier for set! in a future upstream PR later. `(set! "language" @foo :downcase)`

Reply:
It could even be a modifier for set! In a future PR `(set! "language" @foo downcase!)` or directly apply on a key `(set! "language" @foo) (downcase! "language")`

Reply:
Thanks for the feedback, I liked the suggestion to allow the directives to be chained, and made the corresponding changes.

```scheme
(#set! "language" @foo) (#downcase! "language")
```